### ISSUE TITLE
temperature: read values from `/sys` instead of parsing `sensors -u`

### ISSUE THREAD

**Issue by The-King-of-Toasters** [greshake/i3status-rust#469]
I thought it was weird that the program was parsing `sensors -u` instead of reading values from `/sys` like `sensors` does. [A `strace`](https://www.sgregoratto.me/paste/sensors.log.txt) shows that it reads values from `/sys/class/hwmon/hwmon4` on my machine at least. We can confirm this with the following script:

```sh
while true; do
	pr -mtw11 <(sensors -u coretemp-isa-0000 | awk '/temp._input/{print $2 * 1000}') \
               <(cat /sys/class/hwmon/hwmon4/temp?_input)
	echo
	sleep 1
done
```

**Comment by The-King-of-Toasters** [comment]
To test this, I wrote a PoC Go that outputs something similar to `sensors -u`

```go
package main

import (
	"fmt"
	"path/filepath"
	"io/ioutil"
	"bytes"
	"strings"
	"strconv"
	"sort"
)

type CoreTemp struct {
	input, max, crit, crit_alarm int
	label string
}

func (c *CoreTemp) Print() {
	fmt.Printf(`%s:
  input: %d.000
  max: %d.000
  crit: %d.000
  crit_alarm: %d.000
`, c.label, c.input, c.max, c.crit, c.crit_alarm)
}

func readTemps(path string) {
	matches, err := filepath.Glob(path + "/temp*")
	if err != nil {
		panic(err)
	}
	sort.Strings(matches)

	ct := CoreTemp{}
	for i, f := range(matches) {
		if (i + 1) % 5 == 0 {
			ct.Print()
		}

		buf, err := ioutil.ReadFile(f)
		if err != nil {
			panic(err)
		}
		// remove trailing newline
		buf = buf[0:len(buf) - 1]

		var field *int
		switch {
		case strings.HasSuffix(f, "input"):
			field = &ct.input
		case strings.HasSuffix(f, "max"):
			field = &ct.max
		case strings.HasSuffix(f, "crit"):
			field = &ct.crit
		case strings.HasSuffix(f, "crit_alarm"):
			field = &ct.crit_alarm
		case strings.HasSuffix(f, "label"):
			ct.label = string(buf)
			fallthrough
		default:
			continue
		}

		num, err := strconv.Atoi(string(buf))
		if err != nil {
			panic(err)
		}
		*field = num / 1000
	}
}

func main() {
	names, err := filepath.Glob("/sys/class/hwmon/hwmon*")
	if err != nil {
		panic(err)
	}
	for _, v := range(names) {
		buf, err := ioutil.ReadFile(v + "/name")
		if err != nil {
			panic(err)
		}
		if bytes.Equal([]byte("coretemp\n"), buf) {
			readTemps(v)
		}
	}
}
```

**Comment by The-King-of-Toasters** [comment]
Note that this is all conditional on me using the [coretemp driver](https://www.kernel.org/doc/Documentation/hwmon/coretemp) for Intel CPUs. You'd have to do some extra heavy lifting to support different sensors.

**Comment by atheriel** [comment]
The `sensors` program absolutely reads from sysfs (I have looked at the source code), and we could in theory bypass it. However, the `hwmon` stuff was pretty underdocumented the last time I looked into it, so I thought it was reasonable to punt on handling that complexity to a well-known program.

> You'd have to do some extra heavy lifting to support different sensors.

Yes, and I'm wondering why we'd gain from doing that work in this case.
